### ISSUE TITLE
Example of importer server files

### ISSUE THREAD

**Issue by lars18th** [avcsa/openepg#2]
Hi,

Please, can you add one example of the files inside the provider server?

I like to try this project, but I don't know how to provide the EIT information. I suspect it uses XMLTV files, but I don't know how to store them on the FTP server.

Thank you!

**Comment by jmrunge** [comment]
Hi. I cannot provide example files of this because they are in a propietary format. But you can take a look at how the parser works and figure it out (or write your own parser, that was the original idea).

**Comment by lars18th** [comment]
Hi @jmrunge ,

> But you can take a look at how the parser works and figure it out (or write your own parser, that was the original idea).

Thank you! However, we need some "example" as starting point.
Please, can you provide some guide?

**Comment by jmrunge** [comment]
Take a look at reportv_dataprovider:

*
https://github.com/avcsa/openepg/blob/4fcc367507b1b96820e03096d424f8ee905c6e0f/importer/reportv_dataprovider.js#L61

Here you can see, the three files (services, programs, schedules) are |
separated text files (like csv, but | instead of ,)


*
https://github.com/avcsa/openepg/blob/4fcc367507b1b96820e03096d424f8ee905c6e0f/importer/reportv_dataprovider.js#L72

Here you can see how services are processed


*
https://github.com/avcsa/openepg/blob/4fcc367507b1b96820e03096d424f8ee905c6e0f/importer/reportv_dataprovider.js#L82

Here you can see how programs are processed


*
https://github.com/avcsa/openepg/blob/4fcc367507b1b96820e03096d424f8ee905c6e0f/importer/reportv_dataprovider.js#L107

And here, the scheds.

Hope you can take this as starting point!

2017-11-07 10:53 GMT-03:00 lars18th <notifications@github.com>:

**Comment by lars18th** [comment]
Hi @jmrunge ,

Thank you! Now I see where to start... However, I appreciate a "fake" three files that illustrates the current format (I know, a "custom" format, however with it will be more easy!).

Regards!

**Comment by jmrunge** [comment]
Totally, the idea behind the dataprovider is that you write your own custom
format... Its quite easy, taking the default reportv_dataprovider file as
an initial start point.  Then, in config file you can tell openepg which
provider to use. If you want, you can do a pull request with your provider!

Regards,

2017-11-07 12:31 GMT-03:00 lars18th <notifications@github.com>:

**Comment by lars18th** [comment]
Hi @jmrunge ,

I have some questions:

I don't know about these values (all others are very clear):

```
    service.comment   = ... ;
    service.signalId  = ... ;
```

I'm not sure about the types of the values.

So, please, can you add some "fake_dataprovider.js" that implements a very simple example?
We appreciate it!

**Comment by jmrunge** [comment]
serviceId is the service id, which is used to reference it from the scheds
file. Comment is a string, could be the name of the service.  Sorry but
really dont have the time now for writing (and testing) a
"fake_dataprovider".

Kind Regards,

2017-11-07 13:38 GMT-03:00 lars18th <notifications@github.com>:

**Comment by lars18th** [comment]
Hi @jmrunge ,

> serviceId is the service id, ...

Yes, I know about "service.serviceID". The question is about **"service.signalID"**. How does it match?

> Sorry but really dont have the time now for writing (and testing) a
> "fake_dataprovider".

I understand. Then please, if you can add some "example" file with this custom format (even if it only has one entry) it will be unvaluable!

Thank you!

**Comment by jmrunge** [comment]
originalNetworkId, transportStreamId and signalId are part of the MPEG
specification and by use of the multiplexor and further more the STB.
If you dont know what to put there perhaps your multiplexer does not use
them. Take note Im putting service Id in all fields because this was my
case.

2017-11-08 4:24 GMT-03:00 lars18th <notifications@github.com>:

**Comment by lars18th** [comment]
Hi @jmrunge ,

I know about DVB descriptors, and originalNetworkID and transportStreamID are very clear.
What represents signalID in your code? Perhaps the "cell ID" ?

My objective it's use your project for injecting own EPG to DVB MPTS delivered using a software muxor and a low cost DVB-T modulator.

**Comment by jmrunge** [comment]
After struggling with the code (didnt remember that part well) I found that
serviceId is the one that is informed as part of the DBV spec (as used by
CherryEGP, the core of OpenEPG) and signalId is the internal code of the
service itself. To be sure everything works I recommend use the same number
on both, because I saw some parts of the code in which signalId is used and
others where serviceId is used to match with the events. Sorry but cant
spend more time to fix this bug right now.

Exactly for this are we using the software! Hope you will be able to make
it fit your needs. Sure some coding will be necessary.

2017-11-08 10:09 GMT-03:00 lars18th <notifications@github.com>:
